### PR TITLE
feat: Add archive flag in table and detail page

### PR DESCRIPTION
**Pull request by killbond007**
## Description

Add archive flag in table and detail page

Capture d’écran 2021-03-29 à 09 26 34
Capture d’écran 2021-03-29 à 09 26 29

### CODE REVIEW

File: src/api-gateway/resolvers/flag-resolver.ts [hunk 1]
```diff
@@ -237,20 +246,24 @@ class UpFlagDetailsArgs {
...
     @Ctx() { model: { flagModel, userWorkspace }, userId }: IContext
   ): Promise<FlagsStatus> {
     await assertWorkspace(userWorkspace, userId, workspaceId);
 
     const query = {
       workspace: workspaceId,
+      archived,
     };
 
     const [total, flags] = await Promise.all([
```
Comment (bitQ2019) [inline src/api-gateway/resolvers/flag-resolver.ts:269]: should add "archived" in the total count query

File: src/shared/flag-details/flag-details.tsx [hunk 2]
```diff
@@ -105,59 +109,97 @@ export function FlagDetails({
...
+                ? "This flag will be restored across all environments."
+                : "This flag will be archived across all environments and will only appear in your list when filtered for."}
+            </Typography.Text>
+            <Popconfirm
+              title={
+                flagDetails?.archived
+                  ? "Are you sure to restore this flag?"
+                  : "Are you sure to archived this flag?"
+              }
+              onConfirm={() =>
```
Comment (bitQ2019) [inline src/shared/flag-details/flag-details.tsx:184]: prefer to use "async await" and add error handling

File: src/shared/flag-details/flag-details.tsx [hunk 3]
```diff
@@ -105,59 +109,97 @@ export function FlagDetails({
...
+            </Form.Item>
+          </Form>
+        </Tabs.TabPane>
+        <Tabs.TabPane tab="Setting" key="2">
+          <Typography.Title level={5} type="secondary">
+            {flagDetails?.archived ? "Restore flag" : "Archive flag"}
+          </Typography.Title>
+          <Space direction="vertical">
+            <Typography.Text>
+              {flagDetails?.archived
```
Comment (bitQ2019) [inline src/shared/flag-details/flag-details.tsx:174]: how about move this to i18n

**Review by bitQ2019 — CHANGES_REQUESTED** [review]
see comments above

File: src/api-gateway/resolvers/flag-resolver.ts (hunk 1 above)
Comment (killbond007) [reply src/api-gateway/resolvers/flag-resolver.ts:269]: il seems that estimatedDocumentCount() does not take a query filter ? when i add archived it's not working , or should we use countDocuments({ workspace: workspaceId, archived }) ?

Comment (bitQ2019) [reply src/api-gateway/resolvers/flag-resolver.ts:269]: seems we have to use "countDocuments" with a query

**Review by bitQ2019 — APPROVED** [review]

**Review by puncsky — APPROVED** [review]

**Pull request merged**